src/clients/c++/request.h

```diff
@@ -616,6 +618,17 @@ class InferContext {
   /// \return Error object indicating success or failure.
   virtual Error AsyncRun(std::shared_ptr<Request>* async_request) = 0;
 
+  /// Similar to AsyncRun() above. However, this function does not return the
```

Comment:
For the above AsyncRun put a deprecation notice at the top of the description.
"DEPRECATED: This function is deprecated and will be removed in a future version of this API. Use AsyncRun(OnCompleteFn)".

On GetReadyAsyncRequest also should put a deprecation notice.
"DEPRECATED: This function is deprecated and will be removed in a future version of this API. This function is only useful with the deprecated version of AsyncRun(). Instead use AsyncRun(OnCompleteFn)"